### ISSUE TITLE
Home Problematic Dropdown Menu Pattern 

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#752]
A toggle control is positioned after the “Edit Title” icon, which when activated allows a dropdown menu to be expanded.  The menu contains the “Duplicate Set” and “Share with Teachers” links.  The implementation of both the menu and toggle control follows the same pattern as previously described in the PageIndex component section of this document.  Unlike in that implementation, however, there is no invalid nesting of components and the toggle control and menu content are at least usable by a screen reader user who is aware of their purpose. 
 
The mark-up of the toggle button and sub-menu content does still present some accessibility concerns, as follows: 
 
* The toggle button has no accessible label.  Visually, it is labelled using a Font Awesome icon, but no textual alternative is implemented. 
* An aria-haspopup attribute (with a value of “true”) has been applied to the toggle button.  This is incorrect because, while the button does control the display of menu-style content, that content itself is not (and should not) be marked up as a menu proper. 
* The menu is contained inside a < div > instead of a < ul >. 
 
Finally, users who are able to operate the menu using a mouse may become confused, when they find that activating the “Share with Teachers” option actually opens the Twitter website with a pre-filled tweet. 
 
Suggested Solution 
 
An accessible label must be applied to the toggle, preferably implemented as visible content inside the < button > so that speech input users can easily expand and collapse the menu.  The text can be something like: “More Options”, but doesn’t need to include the name of the current problem set because multiple such buttons are not being shown in this context.  The label string will need to be localized. 
 
Next, Unnecessary and problematic attributes should be removed from all < button > and < a > controls.  This includes the aria-haspopup attribute from the toggle button, as well as the tabindex and type attributes from all other < button > and < a > elements. 
 
Thirdly, the sub-menu should be marked up as an unordered list instead of a flat < div >, with the two child links wrapped inside < li > elements. 
 
And finally, it would be helpful to modify the link text of the “Share with Teachers” option so that it more accurately reflects its true purpose and destination.  For example, the text could just read “Share on Twitter”.  This change will require  updating the share_with_teachers key in the strings.js file.

**Comment by johnhbenetech** [comment]
Proposed Code Sample  
File: src/components/Home/index.js 

`<div className="dropdown">  <button className="btn dropdown-toggle" id="dropdownMenuButton" datatoggle="dropdown" aria-expanded="false">   {Locales.strings.more_options}   <FontAwesome className={classNames('fa-2x')} name="ellipsis-v" />  </button>  <ul className="dropdown-menu" aria-labelledby="dropdownMenuButton">   {params.action === 'edit' && (    <React.Fragment>     <li>      <a       href="#"       className="dropdown-item"       onClick={this.props.duplicateProblemSet} 
onKeyPress={this.props.duplicateProblemSet}      >       {Locales.strings.duplicate_set}       <FontAwesome size="lg" name="copy" />      </a>     </li>     <li>      <a       href="#"       className="dropdown-item"       onClick={this.shareOnTwitter}       onKeyPress={this.shareOnTwitter}      >       {Locales.strings.share_with_teachers}       <FontAwesome size="lg" name="twitter"/>      </a>     </li>    </React.Fragment>   )}  </ul> </div>
`

File: src/strings.js 
 
More_options: 'More Options',

**Comment by alexrcabral** [comment]
The dropdown is now more easy to access to use on VoiceOver, but I think we should implement the "Share on Twitter" change as well. (It currently still reads as "Share with Teachers")

**Comment by rupeshparab** [comment]
I have updated the text

**Comment by jscholes** [comment]
@rupeshparab - the `dropdownMenuButton` control still has an `aria-haspopup` attribute which needs to be removed.

In addition, the first option within the dropdown is followed by the text, "Opens in new tab", while the second has "Opens in new window".  But a page cannot guarantee the difference between the two.  In chrome, the button opens a new tab, while Firefox shows an alert about a blocked pop-up window.

As such, we don't need these two entries in strings.js and should just decide on using one or the other.  "Opens in new window" is standard and recognisable.

**Comment by sinabahram** [comment]
Small note, but “open new tab” is a bit shorter, both in speech and Braille, and most of the time true, I’d say, so I’d vote for that over window, but again, this is a tiny note.

**Comment by jscholes** [comment]
@rupeshparab - great.  Could we add an S to the "open new tab" string ("opens new tab")?

**Comment by rupeshparab** [comment]
sure, I have changed open to opens

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/885aae52f87617f4159e287d84f10615667dc81e.
